src/Microsoft.DotNet.ScenarioTests.SdkTemplateTests/SdkTemplateTests.cs

```diff
@@ -72,6 +72,112 @@ public void VerifyMSTestTemplate(DotNetLanguage language)
             DotNetSdkActions.Test);
         newTest.Execute(_sdkHelper, _scenarioTestInput.TestRoot);
     }
+    
+    [Theory]
+    [InlineData(DotNetLanguage.CSharp)]
+    public void VerifyCSharpWPFTemplate(DotNetLanguage language)
+    {
+        var newTest = new SdkTemplateTest(
+            nameof(SdkTemplateTest), language, _scenarioTestInput.TargetRid, DotNetSdkTemplate.Wpf,
+            DotNetSdkActions.Test | DotNetSdkActions.Publish);
+        newTest.Execute(_sdkHelper, _scenarioTestInput.TestRoot);
+    }
+    
+    [Theory]
+    [InlineData(DotNetLanguage.VB)]
+    public void VerifyVBWPFTemplate(DotNetLanguage language)
+    {
+        var newTest = new SdkTemplateTest(
+            nameof(SdkTemplateTest), language, _scenarioTestInput.TargetRid, DotNetSdkTemplate.Wpf,
+            DotNetSdkActions.Test | DotNetSdkActions.Publish);
+        newTest.Execute(_sdkHelper, _scenarioTestInput.TestRoot);
+    }
+    
+    [Theory]
+    [InlineData(DotNetLanguage.CSharp)]
+    public void VerifyWebAppTemplate(DotNetLanguage language)
+    {
+        var newTest = new SdkTemplateTest(
+            nameof(SdkTemplateTest), language, _scenarioTestInput.TargetRid, DotNetSdkTemplate.WebApp,
+            DotNetSdkActions.Test | DotNetSdkActions.PublishComplex);
+        newTest.Execute(_sdkHelper, _scenarioTestInput.TestRoot);
+    }
+    
+    
+    [Theory]
+    [InlineData(DotNetLanguage.CSharp)]
+    public void VerifyCSharpWinformsTemplate(DotNetLanguage language)
+    {
+        var newTest = new SdkTemplateTest(
+            nameof(SdkTemplateTest), language, _scenarioTestInput.TargetRid, DotNetSdkTemplate.Winforms,
+            DotNetSdkActions.Test | DotNetSdkActions.Publish);
+        newTest.Execute(_sdkHelper, _scenarioTestInput.TestRoot);
+    }
+    
+    [Theory]
+    [InlineData(DotNetLanguage.VB)]
+    public void VerifyVBWinformsTemplate(DotNetLanguage language)
+    {
+        var newTest = new SdkTemplateTest(
+            nameof(SdkTemplateTest), language, _scenarioTestInput.TargetRid, DotNetSdkTemplate.Winforms,
+            DotNetSdkActions.Test | DotNetSdkActions.Publish);
+        newTest.Execute(_sdkHelper, _scenarioTestInput.TestRoot);
+    }
+    
+    [Theory]
+    [InlineData(DotNetLanguage.CSharp)]
+    public void VerifyReferenceInConsoleTemplate(DotNetLanguage language)
```

Comment:
For templates that only support a single language, you could use `[Fact]` instead of `[Theory]`. Facts are just tests that take no input parameters.

Reply:
While this specific test now has multiple languages, I've followed your suggestion here for tests that can't use the variables such as WebApp